src/redis_lock/__init__.py

```diff
@@ -313,43 +323,9 @@ def release(self):
         else:
             self._held = False
 
-
-class InterruptableThread(threading.Thread):
-    """
-    A Python thread that can be requested to stop by calling request_exit()
-    on it.
-
-    Code running inside this thread should periodically check the
-    `should_exit` property (or use wait_for_exit_request) on the thread
-    object and stop further processing once it returns True.
-    """
-    def __init__(self, *args, **kwargs):
-        self._should_exit = threading.Event()
-        super(InterruptableThread, self).__init__(*args, **kwargs)
-
-    def request_exit(self):
-        """
-        Signal the thread that it should stop performing more work and exit.
-        """
-        self._should_exit.set()
-
-    @property
-    def should_exit(self):
-        return self._should_exit.isSet()
-
-    def wait_for_exit_request(self, timeout=None):
-        """
-        Wait until the thread has been signalled to exit.
-
-        If timeout is specified (as a float of seconds to wait) then wait
-        up to this many seconds before returning the value of `should_exit`.
-        """
-        should_exit = self._should_exit.wait(timeout)
-        if should_exit is None:
-            # Python 2.6 compatibility which doesn't return self.__flag when
-            # calling Event.wait()
-            should_exit = self.should_exit
-        return should_exit
+    def __del__(self):
+        if self._lock_renewal_thread is not None:
+            self._stop_lock_renewer()
```

Comment:
One interesting thing here is that `self._stop_lock_renewer()` will potentially block. Not sure if that's desirable during GC. 

Maybe don't implement `__del__` at all? It don't really make sense to have `__del__` if we use the weakref (it does the same thing but better).